app/models/encounter.rb

```diff
@@ -0,0 +1,42 @@
+class Encounter < ApplicationRecord
+  belongs_to :patient
+  belongs_to :facility
+
+  has_many :encounter_events
+  has_many :blood_pressures, through: :encounter_events, source: :encounterable, source_type: 'BloodPressure'
+
+  def events
+    encounter_events.includes(:encounterable).map(&:encounterable)
+  end
+
+  def self.create_encounter_with_events!(params)
```

Comment:
Can you outline the use case for this method?

In my mind, you'd simply create a `BloodPressure`. Perhaps that model can be "observable" and, before create, finds or creates an encounter for the patient. The logic wouldn't live in Encounter at all.

Reply:
This method and its uses are still under progress and it will support editing as well (find or create). But generally speaking, the purpose of this method would be to create an Encounter "as-is" (sent from the app). The app will send us an appropriate Encounter ID and its constituent `Observations` since the data will now be nested within the Encounter object.

The API will also represent a completely new sync resource called `Encounter` and that will simply linearly traverse the contents of the Encounter payload.

I don't think I understood why we need to approach creating Encounters from bottom-up (through BloodPressures / PrescriptionDrugs) on a modelling level. This would be true for backwards-compatability's for existing BP/PD APIs sake but the primary Encounter API needn't behave that way.

Reply:
Haven't fully thought through this method yet, subject to change, but the basic intention should be:

Transactionally create:
- The actual `BloodPressure`
- Associate `BloodPressure` against the `Observation` table
- Associate the Observation with the appropriate `Encounter`

Does not support partial creates for the Encounter (does not allow saving 5/10 BPs). It's either 100% updated or the whole thing is unsynced.